contracts/src/fp/OPSuccinctPermissionedFaultDisputeGame.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "forge-std/console.sol";
+
+// Contracts
+import {OPSuccinctFaultDisputeGame} from "src/fp/OPSuccinctFaultDisputeGame.sol";
+
+// Libraries
+import {Duration, GameStatus} from "src/dispute/lib/Types.sol";
+import {BadAuth} from "src/dispute/lib/Errors.sol";
+
+// Interfaces
+import {ISP1Verifier} from "@sp1-contracts/src/ISP1Verifier.sol";
+import {IDisputeGameFactory} from "src/dispute/interfaces/IDisputeGameFactory.sol";
+
+/// @title OPSuccinctPermissionedFaultDisputeGame
+/// @notice OPSuccinctPermissionedFaultDisputeGame is a contract that inherits from `OPSuccinctFaultDisputeGame`, and contains two roles:
+///         - The `challenger` role, which is allowed to challenge a game.
+///         - The `proposer` role, which is allowed to create proposals and participate in their game.
+///         This contract exists as a way for networks to support the fault proof iteration of the OptimismPortal
+///         contract without needing to support a fully permissionless system. Permissionless systems can introduce
+///         costs that certain networks may not wish to support. This contract can also be used as a fallback mechanism
+///         in case of a failure in the permissionless fault proof system in the stage one release.
+contract OPSuccinctPermissionedFaultDisputeGame is OPSuccinctFaultDisputeGame {
+    /// @notice An address that is allowed to challenge games.
+    address internal immutable CHALLENGER;
+    /// @notice An address that is allowed to propose games.
+    address internal immutable PROPOSER;
+
+    modifier onlyChallenger() {
+        if (msg.sender != CHALLENGER) {
+            revert BadAuth();
+        }
+        _;
+    }
+
+    modifier onlyProposer() {
+        if (msg.sender != PROPOSER) {
+            revert BadAuth();
+        }
+        _;
+    }
```

Comment:
One concern I have with this design is that in most practical settings to reach Stage 1, you want to have multiple challengers and proposers. 

Do you think it's better to have a pass-through contract where games can be created where this logic can be validated?

Or do you want to allow passing arrays for both of these arguments?

My worry with this design is that it's not easily extensible.